crates/forge/tests/e2e/running.rs

```diff
@@ -414,6 +414,48 @@ fn with_include_ignored_flag_and_filter() {
     );
 }
 
+#[test]
+fn with_rerun_failed_flag() {
```

Comment:
We should also have a test where there's no `.prev_tests_failed` initially and we call runner two times, first normally and second with `--rerun-failed` to see if it's created correctly.

This is relying on manually prepared cache file which is good, but not exclusively.

Reply:
sure, but it's related to this [comment]( https://github.com/foundry-rs/starknet-foundry/pull/1084#discussion_r1392209904 ). After making a decision, I will add a new tests